### ISSUE TITLE
Node 'training/Adam/gradients/gradients/batch_normalization_8/cond_grad/StatelessIf': Connecting to invalid output 3 of source node batch_normalization_8/cond which has 3 outputs. Try using tf.compat.v1.experimental.output_all_intermediates(True).

### ISSUE THREAD

**Issue by Hjiwnain** [tensorflow/tensorflow#54222]
<em>Please make sure that this is an issue related to keras.
tag:keras_template</em>

**Important Notice**

Please note that `tf.keras` code was moved entirely to
[keras-team/keras](https://github.com/keras-team/keras) repository

You can open any code/doc bugs, performance issues, and feature requests
 in [keras-team/keras](https://github.com/keras-team/keras/issues) repository

`tf.keras` related issues opened in
[tensorflow/tensorflow](https://github.com/tensorflow/tensorflow) repository may
not get attention as [keras-team/keras](https://github.com/keras-team/keras)
repository is dedicated for the development of `keras` code

**Comment by tilakrayal** [comment]
@Hjiwnain ,
In order to expedite the trouble-shooting process, could you please provide a minimal code snippet and the TensorFlow version you are using.

**Comment by tilakrayal** [comment]
Also please take a look at this [link](https://www.tensorflow.org/api_docs/python/tf/compat/v1/experimental/output_all_intermediates) for the information regarding tf.compat.v1.experimental.output_all_intermediates.It helps.Thanks!
